mono/mini/mini-runtime.c

```diff
@@ -4472,8 +4472,9 @@ mini_init (const char *filename, const char *runtime_version)
 
 #ifdef MONO_ARCH_EMULATE_FREM
 // Wrapper to avoid taking address of overloaded function.
-static double
-mono_fmod (double a, double b)
+G_EXTERN_C double mono_fmod (double a, double b);
```

Comment:
shouldn't this still be `static`?

Reply:
I am not sure. Encountering other statics is why I stopped adding all `register_icall` parameters.
I'm also not sure about this:
```
register_icall (mono_gc_wbarrier_generic_nostore, "wb_generic", "void ptr", FALSE);
```

If they were all static, and the names all matched, I would have added all `register_icall` parameters. Why do most of the names match but not this one?

Reply:
Yeah, if it worked before w/ static, it should still.
I should probably get the rest of these:
 git grep register_icall | grep \" |wc -l
185.

Reply:
1. > Encountering other statics is why I stopped adding all register_icall parameters.
  
   I don't understand what you mean.
   My understanding was that icalls should be extern C because the runtime uses a C calling convention to invoke them.  whether the icall is static or not shouldn't matter.
2. the "name" of the icall basically doesn't matter for jit icalls (as long as it's used consistently). it's just a key into a hash.

   There are two ways to lookup an icall: by C function ptr address (method-to-ir does this in the `MONO_CEE_MONO_ICALL` case via `mono_find_jit_icall_by_addr`) or by "name" (which is what, for example, `mono_resolve_patch_target` does via `mono_find_jit_icall_by_name`).

Reply:
It isn't calling convention, it is the linker to find the name.

There were a bunch of link errors, on some of these names.

There really isn't such a thing as multiple calling conventions -- only NT/x86 made that mistake, of having more than one. Otherwise all functions have the same calling convention and it can be ignored. NT/amd64 also somewhat with `vectorcall`.

https://jenkins.mono-project.com/job/test-mono-pull-request-amd64-osx-products-sdks-ios/4452/consoleFull

```
<span class="timestamp"><b>02:52:08</b> </span>clang -ObjC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk -miphoneos-version-min=10.1 -arch arm64 -framework Foundation -framework UIKit -o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/bin/ios-dev/test-Mono.Runtime.Tests.app/test-Mono.Runtime.Tests /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/mscorlib.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/System.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/System.Xml.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/System.Core.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/I18N.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/I18N.West.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/Mono.Simd.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/Mono.Security.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/System.Numerics.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/System.Numerics.Vectors.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/nunitlite.dll.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/test-runner.exe.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/aot-cache/monotouch_Mono.Runtime.Tests_test.dll.o main.o /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/out/ios-target64-release/lib/libmonosgen-2.0.a /Users/builder/jenkins/workspace/test-mono-pull-request-amd64-osx-products-sdks-ios/sdks/ios/runtime/libmonoios.a -liconv
<span class="timestamp"><b>02:52:08</b> </span>Undefined symbols for architecture arm64:
<span class="timestamp"><b>02:52:08</b> </span>  "_mono_aot_init_gshared_method_mrgctx", referenced from:
<span class="timestamp"><b>02:52:08</b> </span>      wrapper_managed_to_native_object___icall_wrapper_mono_aot_init_gshared_method_mrgctx_intptr_intptr_intptr in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>  "_mono_aot_init_gshared_method_this", referenced from:
<span class="timestamp"><b>02:52:08</b> </span>      wrapper_managed_to_native_object___icall_wrapper_mono_aot_init_gshared_method_this_intptr_intptr_object in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>  "_mono_aot_init_gshared_method_vtable", referenced from:
<span class="timestamp"><b>02:52:08</b> </span>      wrapper_managed_to_native_object___icall_wrapper_mono_aot_init_gshared_method_vtable_intptr_intptr_intptr in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>  "_mono_aot_init_llvm_method", referenced from:
<span class="timestamp"><b>02:52:08</b> </span>      wrapper_managed_to_native_object___icall_wrapper_mono_aot_init_llvm_method_intptr_intptr in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>  "_mono_create_corlib_exception_0", referenced from:
<span class="timestamp"><b>02:52:08</b> </span>      Crimson_CommonCrypto_Cryptor_Create_Crimson_CommonCrypto_CCOperation_Crimson_CommonCrypto_CCAlgorithm_Crimson_CommonCrypto_CCOptions_byte___byte__ in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      Microsoft_Win32_RegistryKey__ctor_Microsoft_Win32_RegistryHive in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      Microsoft_Win32_RegistryKey_CreateSubKey_string in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      Microsoft_Win32_RegistryKey_CreateSubKey_string_bool in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      Microsoft_Win32_RegistryKey_CreateSubKey_string_bool_Microsoft_Win32_RegistryOptions in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      Microsoft_Win32_RegistryKey_CreateSubKey_string_Microsoft_Win32_RegistryKeyPermissionCheck in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      Microsoft_Win32_RegistryKey_CreateSubKey_string_Microsoft_Win32_RegistryKeyPermissionCheck_Microsoft_Win32_RegistryOptions in mscorlib.dll.o
<span class="timestamp"><b>02:52:08</b> </span>      ...
```

Reply:
Maybe  it is `mono_emit_jit_icall`?